Stac-25353: Write concise 'create an integration' page and add stackpacks reference (#364)
rancher/stackstate-product-docs: docs/latest/modules/en/pages/setup/custom-integrations/create.adoc

ifdef::ss-ff-stackpacks2_enabled[]
= Tutorial: Create a Custom Integration (StackPack)
:page-languages: [en, de, es, fr, ja, pt, zh]
:revdate: 2026-01-08
:page-revdate: {revdate}
:description:  Develop a Custom Integration (StackPack)

== Goal

> As a devops developer, you want to customize SUSE Observability to monitor your own applications you have deployed.

Reviewer: ```suggestion
As a devops developer, you want to customize {stackstate-product-name} to monitor your own applications you have deployed.
```

[NOTE]
====

> This page is about developing a custom integration. Installing an already built integration is described here: xref:/setup/[TODO]

Reviewer: ```suggestion
This page is about developing a custom integration. For information on installing an already built integration, see xref:/setup/[TODO].
```

====

== Prerequisites

- Access to the xref:/setup/cli/cli-sts.adoc[SUSE Observability CLI]

> - Access to a running SUSE Observability instance xref:/setup/install[TODO]

Reviewer: ```suggestion
- Access to a running {stackstate-product-name} instance xref:/setup/install[TODO]
```

=== Scaffold a New StackPack

> The `scaffold` subcommand creates a new StackPack project structure from a template, getting you started with all concepts that are relevant.

Reviewer: ```suggestion
The `scaffold` subcommand creates a new StackPack project structure from a template, getting you started with all relevant concepts.
```

> - Create a new StackPack project using the CLI:

Reviewer: ```suggestion
Create a new StackPack project using the CLI:
```

+
[source,bash]
----
sts stackpack scaffold --name my-stackpack
...
...

✅ ✓ Scaffold complete!

Next steps:
1. Review the generated files in: ./my-stackpack
2. Check the ./my-stackpack/README.md for instructions on what to do next.
----

=== Test Deploy StackPack

> The `test` subcommand command that streamlines the stackpack development workflow by automating the package → upload → install/upgrade sequence with automatic version management for testing iterations.

Reviewer: ```suggestion
The `test` subcommand command streamlines the stackpack development workflow by automating the package → upload → install/upgrade sequence with automatic version management for testing iterations.
```

> - Rapidly test your StackPack in a pre-production environment (requires a running Suse Observability instance):

Reviewer: ```suggestion
- Rapidly test your StackPack in a pre-production environment (requires a running {stackstate-product-name} instance):
```
